apps/backend/src/authn/apikey.strategy.ts

```diff
@@ -0,0 +1,24 @@
+import {Injectable} from '@nestjs/common';
+import {PassportStrategy} from '@nestjs/passport';
+import HeaderAPIKeyStrategy from 'passport-headerapikey';
+import {ApiKeyService} from '../apikeys/apikey.service';
+import {User} from '../users/user.model';
+
+@Injectable()
+export class APIKeyStrategy extends PassportStrategy(
+  HeaderAPIKeyStrategy,
+  'apikey'
+) {
+  constructor(private readonly apiKeyService: ApiKeyService) {
+    super(
+      {header: 'Authorization', prefix: 'Api-Key '},
+      false,
+      async (
+        apikey: string,
+        done: (arg1: null, arg2: Promise<User | null> | boolean) => unknown
+      ) => {
+        return done(null, this.apiKeyService.checkKey(apikey));
```

Comment:
instead of setting error to null, this should be set to an unauthorized exception that is raised based on checking the return value of `checkKey` function. This would probably remove the need to have a `handleRequest` method in the guard. If you take a look at local strategy you can see this design in action.

Reply:
Returning an error to to the callback does not cause a request exception, the request just continues with a null User. However, throwing exceptions inside of `checkKey` does seem to cause a request exception.